app/controllers/api/v1/keywords_controller.rb

```diff
@@ -12,11 +12,38 @@ def index
         render json: KeywordSerializer.new(keywords_list, pagy_options(pagy))
       end
 
+      def create
+        if csv_form.save(create_params[:file])
+          render json: create_success_response
+
+        else
+          render_errors(
+            details: csv_form.errors.full_messages,
+            code: :invalid_file,
+            status: :unprocessable_entity
+          )
+        end
+      end
+
       private
 
       def keywords
         KeywordsQuery.new(current_user).call
       end
+
+      def csv_form
+        @csv_form ||= CSVUploadForm.new(current_user)
```

Comment:
I think memoization will not work here, every time a request comes, it will not have the previous request information(unless it is stored somewhere).  What do you think?
```suggestion
  CSVUploadForm.new(current_user)
```

Reply:
Ahh, I see you are accessing it in `render_errors` 👍